packages/compass-connections/src/provider.ts

```diff
@@ -12,8 +12,16 @@ export { useConnections } from './stores/connections-store';
 export { useConnectionRepository } from './hooks/use-connection-repository';
 export { useActiveConnections } from './hooks/use-active-connections';
 
+class TestConnectionsManager extends EventEmitter {
+  getDataServiceForConnection() {
+    return new EventEmitter() as unknown as DataService;
+  }
+}
+
 const ConnectionsManagerContext = createContext<ConnectionsManager | null>(
-  null
+  process.env.NODE_ENV === 'test'
+    ? (new TestConnectionsManager() as unknown as ConnectionsManager)
+    : null
```

Comment:
We were doing this also in the locator. Was there a case, I missed that we are capturing with this?

Reply:
Discussed f2f: basically we do return a real connection manager from the locator / hook in test env, but test manager is different as it will provide a fake data service also in cases where we don't really care about these at all